enterprise/backend/src/metabase_enterprise/workspaces/types.clj

```diff
@@ -31,3 +31,123 @@
    [:table [:map
             [:name :string]
             [:schema {:optional true} [:maybe :string]]]]])
+
+;;; ---------------------------------------- Graph/Problem Types ----------------------------------------
+
+;; Node types used in workspace graphs and problem detection
+;; Note: :global-transform is for transforms completely outside the workspace (not enclosed, not checked out)
+(mr/def ::node-type [:enum :input-table :external-transform :workspace-transform :global-transform])
+
+(mr/def ::table-coord
+  "Logical reference to a table by database, schema, and name."
+  [:map
+   [:db_id ::appdb-id]
+   [:schema [:maybe :string]]
+   [:table :string]])
+
+(mr/def ::node-ref
+  "Reference to a node in the graph or a related entity."
+  [:map
+   [:type ::node-type]
+   [:id [:or ::appdb-id ::ref-id]]
+   [:name {:optional true} :string]])
+
+;;; ---------------------------------------- Problem Types ----------------------------------------
+
+;; Problem type definitions with metadata for FE display and merge-blocking behavior
+;; See validation.clj docstring for full documentation table
+
+(def problem-types
+  "All problem types with their metadata.
+
+   Each entry has:
+   - :description - human-readable explanation
+   - :severity - :error, :warning, or :info
+   - :blocks-merge? - whether this problem should prevent merging
+   - :static? - true if determinable from metadata alone, false if requires runtime info
+
+   ## Categories
+
+   Dependency-based (about output state) - prefix indicates who depends on the output:
+   - unused-*              : nothing depends on this output
```

Comment:
instead of prefix wdyt about namespaced?

Reply:
I went with prefix because we'll share these with the frontend, and by default I think namespaces get stripped when converting to JSON? 

I guess we could turn them into qualified names. Certainly more readible in Clojure land with namespaces. Will run it past FE guys.

Reply:
OK they're fine with it, so lets do it! Thanks for the suggestion 💖